src/v4.js

```diff
@@ -2,10 +2,10 @@ import rng from './rng.js';
 import bytesToUuid from './bytesToUuid.js';
 
 function v4(options, buf, offset) {
-  const i = (buf && offset) || 0;
+  const start = (buf && offset) || 0;
```

Comment:
Define this variable only in the `if(buf)` scope below, it's not needed outside. It then simplifies to `offset | 0`.

An alternative would be to use default arguments. We'd have to check how efficient the bable'd code is then though (in terms of bundlesize).